Userland/Services/SpiceAgent/MessageHeader.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023, Caoimhe Byrne <caoimhebyrne06@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Message.h"
+#include <AK/ByteBuffer.h>
+#include <AK/Format.h>
+#include <AK/Forward.h>
+
+namespace SpiceAgent {
+
+// An incoming or outgoing message header
+// This contains information about the message, like how long it is, the type, etc.
+class MessageHeader {
+public:
+    // Used for sending a message to the server
+    static ErrorOr<MessageHeader> create(Message::Type type, u32 data_size, u32 protocol_version = AGENT_PROTOCOL, u64 opaque = 0);
```

Comment:
Do we need a fallible constructor? There isn't anything fallible that happens there.